### PR TITLE
changed: vendor pybind11 using FetchContent

### PR DESCRIPTION
**Pull request by akva2**
This is an alternative to https://github.com/OPM/opm-common/pull/3666

Rather than hosting the source in-tree, vendor it using the FetchContent mechanism .

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -269,10 +269,12 @@ if (OPM_ENABLE_PYTHON)
...
   find_package(pybind11 2.2 CONFIG)
   if (NOT pybind11_FOUND)
-      add_subdirectory(python/pybind11)
-      if (NOT pybind11_INCLUDE_DIRS)
-        set(pybind11_INCLUDE_DIRS ${PYBIND11_INCLUDE_DIR} ${PYTHON_INCLUDE_DIRS})
-      endif()
+      include(FetchContent)
+      FetchContent_Declare(pybind11
+                           URL https://github.com/pybind/pybind11/archive/refs/tags/v2.11.0.tar.gz
+                           URL_HASH SHA512=10edfdc499ed70234de0796f79b9f3a7a14d4baf01fb09fedbd9126e0a13d9cdeb213950c53a8f7a67cd6049b1db31efacfe2192098236b242ad3685967ea907
```
Comment (hakonhagland) [inline CMakeLists.txt:275]: Cool, I like this idea. Then we do not have to manually update the source tree with hundreds of file each time we want to upgrade pybind11.

When I tried this with cmake, I ran into this issue https://stackoverflow.com/a/75007891/2173773. I am using cmake version 3.22.1. The issue can be fixed by putting `DOWNLOAD_EXTRACT_TIMESTAMP ON` before the URL and URL_HASH properties (tested on Ubuntu 22.04).

**Review by bska — APPROVED** [review]
I like it, and I think we should do this.  We could possibly use this strategy for [fmtlib](https://github.com/OPM/opm-common/tree/3ff7fffb045c88c5438cd00ac47d931d1fd50154/external/fmtlib) and [cJSON](https://github.com/OPM/opm-common/tree/3ff7fffb045c88c5438cd00ac47d931d1fd50154/external/cjson) too.  Looking at [the documentation](https://cmake.org/cmake/help/latest/module/FetchContent.html), it appears that the [`FETCHCONTENT_BASE_DIR`](https://cmake.org/cmake/help/latest/module/FetchContent.html#variables) variable will address [my earlier concern](https://github.com/OPM/opm-simulators/pull/4849#pullrequestreview-1615509469) about multiple downloads without using system packages, since `pybind11` is header-only.  On the other hand the installed size of `pybind11` is 2MB on my machine so not really a huge deal in the scope of things.

**Comment by akva2** [comment]
jenkins build this please

File: CMakeLists.txt (hunk 1 above)
Comment (akva2) [reply CMakeLists.txt]: moved.

**Comment by akva2** [comment]
jenkins build this please

**Comment by akva2** [comment]
jenkins build this please

**Pull request merged**